internal/fetch/fetch_test.go

```diff
@@ -540,27 +540,43 @@ func TestDownloadTarballEmptySha(t *testing.T) {
 
 func TestLatestCommitAndChecksum(t *testing.T) {
 	const (
-		expectedCommit          = "testcommit123"
-		expectedTarballContents = "mock tarball content for checksum"
-		testOrg                 = "testorg"
-		testRepo                = "testrepo"
+		expectedMasterCommit          = "testcommit123"
+		expectedMasterTarballContents = "mock tarball content for master commit checksum"
+		expectedBranchCommit          = "testothercommit123"
+		expectedBranchTarballContents = "mock tarball content for other branch commit checksum"
+		testOrg                       = "testorg"
+		testRepo                      = "testrepo"
+		testBranch                    = "testbranch"
 	)
 	// Calculate the expected SHA256 for the tarball contents.
 	hasher := sha256.New()
-	hasher.Write([]byte(expectedTarballContents))
+	hasher.Write([]byte(expectedMasterTarballContents))
 	expectedTarballSHA256 := fmt.Sprintf("%x", hasher.Sum(nil))
 
+	hasher.Reset()
+	hasher.Write([]byte(expectedBranchTarballContents))
+	expectedBranchTarballSHA256 := fmt.Sprintf("%x", hasher.Sum(nil))
+
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
```

Comment:
I would factor out `w.WriteHeader(http.StatusOK)` because it is common to every switch statement